lntest/itest/lnd_open_channel_test.go

```diff
@@ -249,6 +250,149 @@ func testOpenChannelAfterReorg(net *lntest.NetworkHarness, t *harnessTest) {
 	closeReorgedChannelAndAssert(t, net, net.Alice, chanPoint, false)
 }
 
+// testOpenChannelFeePolicy checks if different channel fee scenarios
+// are correctly handled when the optional channel fee parameters
+// baseFee and feeRate are provided. If the OpenChannelRequest is not
+// provided with a value for baseFee/feeRate the expectation is that the
+// default baseFee/feeRate is applied.
+// 1.) no params provided to OpenChannelRequest
+//		ChannelUpdate --> defaultBaseFee, defaultFeeRate
+// 2.) only baseFee provided to OpenChannelRequest
+//		ChannelUpdate --> provided baseFee, defaultFeeRate
+// 3.) only feeRate provided to OpenChannelRequest
+//		ChannelUpdate --> defaultBaseFee, provided FeeRate
+// 4.) baseFee and feeRate provided to OpenChannelRequest
+//		ChannelUpdate --> provided baseFee, privided feeRate
+func testOpenChannelUpdateFeePolicy(net *lntest.NetworkHarness,
```

Comment:
That is a very good catch. I think in that case we would land in the `default:` branch of our suggested `switch` statement above since we were not storing any fees(provided or default) in the db and `ourPolicy` might not be populated. This edge case would obviously be solved by setting the default fees in `newChanAnnouncement` but that would defeat the purpose of this change.

Reply:
in this case we have two options:
- A) DB migration, setting the defaults for the channels that are currently pending
- B) if we do not know about that channel, use the defaults


I think we want to catch the cases in B) as "something went wrong here" but I would avoid A) because a migration does not look worth it here.

I would go with B, add a log so we can track what is going on in case someone finds something broken and switch to cancel if we do not have `storedFwdPolicy` in the future.

Reply:
If we go with B) it would mean that we wouldn't return an error in `newChanAnnouncement` if both `ourPolicy` and `storedFwdingPolicy` are `nil`, but that we only log this scenario.